### PR TITLE
A/B test friendly positioning

### PR DESCRIPTION
**Pull request by dracos**
For #1341. Will squash together before merging.

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/FixMyStreet.pm [hunk 1]
```diff
@@ -59,5 +59,26 @@ sub get_country_for_ip_address {
...
+    return 1 unless $self->{c}->user;
+
+    my $reports = $self->{c}->model('DB::Problem')->search({
+        user_id => $self->{c}->user->id,
+        extra => { like => '%gender%' }
+    });
+
+    while (my $report = $reports->next) {
+        my $gender = $report->get_extra_metadata('gender');
+        return 0 if $gender =~ /female|male|other/;
```
Comment (davea) [inline perllib/FixMyStreet/Cobrand/FixMyStreet.pm]: Should `unknown` be included here? Currently a user who clicks 'prefer not to say' will be asked the question every time they make a report.

File: perllib/FixMyStreet/Cobrand/FixMyStreet.pm [hunk 2]
```diff
@@ -59,5 +59,26 @@ sub get_country_for_ip_address {
...
+
+sub ask_gender_question {
+    my $self = shift;
+
+    return 1 unless $self->{c}->user;
+
+    my $reports = $self->{c}->model('DB::Problem')->search({
+        user_id => $self->{c}->user->id,
+        extra => { like => '%gender%' }
+    });
```
Comment (davea) [inline perllib/FixMyStreet/Cobrand/FixMyStreet.pm]: It'd be marginally faster, for users with lots of reports, to use `order_by => { -desc => 'id' }` here.

File: templates/web/fixmystreet.com/report/new/extra_name.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+[%# Same as parent, with added gender question %]
+
+[% IF c.cobrand.ask_gender_question %]
+<p>
+We’re conducting research to make sure our site is good for everyone;
+please help us out by providing your gender:
```
Comment (davea) [inline templates/web/fixmystreet.com/report/new/extra_name.html]: We should add a note making it clear that their answer will not be shared with the council or displayed publicly. Perhaps a link to a longer explanation of why we're asking this on the FAQ or privacy page?

**Comment by davea** [comment]
It's very yellow! The NIU theme really stuck with you eh, @zarino? :wink:

I think it could use a couple of little improvements, as per the inline comments, but otherwise :+1:

File: templates/web/fixmystreet.com/report/new/extra_name.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+</p>
+
+<p class="radio-segmented-control" style="font-size: 80%">
```
Comment (davea) [inline templates/web/fixmystreet.com/report/new/extra_name.html]: This is included on the update form too, but any answer given by the user is discarded. Perhaps this shouldn't be shown on the update form?

**Comment by zarino** [comment]
Yellow is the colour of kings. (And queens.)

File: templates/web/fixmystreet.com/report/new/extra_name.html (hunk 4 above)
Comment (dracos) [inline templates/web/fixmystreet.com/report/new/extra_name.html]: Whoops! Thanks for spotting.

File: templates/web/fixmystreet.com/report/new/extra_name.html (hunk 3 above)
Comment (dracos) [inline templates/web/fixmystreet.com/report/new/extra_name.html]: Yes, good point.

File: perllib/FixMyStreet/Cobrand/FixMyStreet.pm (hunk 1 above)
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/FixMyStreet.pm]: Thought they might change their mind in future, but you're right, it'd be annoying.

**Pull request merged**